protocols/relay/src/copy_future.rs

```diff
@@ -150,95 +157,59 @@ fn forward_data<S: AsyncBufRead + Unpin, D: AsyncWrite + Unpin>(
 
 #[cfg(test)]
 mod tests {
-    use super::CopyFuture;
+    use super::{forward_data, CopyFuture};
     use futures::executor::block_on;
-    use futures::io::{AsyncRead, AsyncWrite};
+    use futures::io::{AsyncRead, AsyncWrite, BufReader, BufWriter};
     use quickcheck::QuickCheck;
     use std::io::ErrorKind;
     use std::pin::Pin;
     use std::task::{Context, Poll};
     use std::time::Duration;
 
-    struct Connection {
```

Comment:
To make confusing diff easier to read:

- `Connection` is only used in `quickckeck` test, thus moved into `quickcheck` test.
- PendingConnection` is only used in `max_circuit_duration` test, thus moved into `max_circuit_duration` test.